app/services/medium_providers/provider.rb

```diff
@@ -24,6 +24,14 @@ def medium_uri
       throw "medium_uri is not implemented for #{self.class.name}"
     end
 
+    # A medium provider can optionally return an array of additional software
+    # URI's to enable during installation, if the template supports it. For
+    # example, for a yum repo to give to Anaconda:
+    #   [{name: 'awesome', baseurl: 'http://yum.example.com/repo/'}]
+    def additional_media
+      []
+    end
```

Comment:
I don't see any possibility to add a GPG key via kickstart `repo` statement. Debian does support that.

I wonder if we should have more formal API here defining name, url, description and gpg key URL.